backend/Dockerfile

```diff
@@ -1,8 +1,23 @@
-FROM python:3.12-slim-bullseye
+FROM python:3.12-slim-bullseye AS poetry
 
-RUN pip install "fastapi[standard]" requests
+RUN pip install poetry
 
 RUN mkdir /app
+WORKDIR /app
+
+COPY pyproject.toml /app
+
+FROM poetry as server
+RUN poetry install --no-root
```

Comment:
You could avoid creating a virtual env in Docker with:

```
RUN poetry config virtualenvs.create false
```

Then you don't need to use poetry anymore when running commands.

Reply:
Could also avoid placing cache in image with:

```
RUN --mount=type=cache,target=/root/.cache \
    poetry install
```

Reply:
yes good idea, fixed in Dockerfile